rfcs/react-components/convergence/global-contex-registration.md

```diff
@@ -0,0 +1,95 @@
+# RFC: Global context registration
+
+@ling1726
+
+## Summary
+
+This RFC proposes that all Fluent UI contexts be registered in the global scope when created. Global scop means
+`window` for browser environments or the `global` object for nodejs environments when using SSR.
+
+## Problem statement
+
+Many of our partners have complex dependency trees. The complexity in dependency management means that duplicate
+packages can exist in `node_modules` folders. Although duplicate depedencies should generally be avoided, this still
+happens quite often due to these factors:
+
+- Propagating dependency bumps downstream can be slow
+- Engineering teams prioritize other work
+- Incorrect version ranges declared
+
+Duplicated code generally is not a huge issue unless there are singletons that are duplicated. React contexts are
+singletons. When duplicated:
+
+```tsx
+<ContextV1.1.Provider>
+
+</ContextV1.1.Provider>
+
+React.useContext(ContextV1.2);
+```
+
+Scenarios like the above can be very difficult to debug. The value of `useContext` will always be the context
+default value because it is not actually wrapped by its provider. The example above makes the reading simpler
+by suffixing the version number in the name. In reality both contexts would actually share the same name even if
+they are different.
+
+The issue has been created before in React's GitHub repository:
+
+- [facebook/react#13346](https://github.com/facebook/react/issues/13346)
+- [reactjs/reactjs.org#1112](https://github.com/reactjs/reactjs.org/pull/1112)
+
+## Detailed Design or Proposal
+
+Prototypes can be found in the following repos for create-react-app and next.js SSR respectively:
```

Comment:
**suggestion:** can we inline that prototype implementation (code) or at least pseudocode to this RFC ? I mean those repos can be removed by any time, thus we would loose reference for future ourselves.

Reply:
In this case I can do it easily - the prototype implem is not that big.